R/postSDA_processing.R

```diff
@@ -13,52 +13,64 @@
 #' @param UMAP.NumEpoc UMAP min. distance. Passed directly to Seurat::RunUMAP()
 #' @import Seurat
 #' @export
-RecreateSeuratObjFromSDAmatrix <- function(cellnames = NULL, recomposedMat = NULL, metaDF, projectName = "SDAproject", dimsToUse = 15, spikeGenes = NULL, nfeatures = 500, 
-                                           tSNE_perplexity = 0.3, UMAP_MinDist = 0.3, UMAP_NumNeib = 35L, UMAP.NumEpoc = 500) {
-  
-  #confirm that gene are rows and columns are cells
-  if(any(grepl("CD1", rownames(recomposedMat)))) {
-    print("genes in rownames, good")
-  } else { recomposedMat <- t(recomposedMat) }
-  
+RecreateSeuratObjFromSDAmatrix <- function (cellnames = NULL, recomposedMat = NULL, metaDF, projectName = "SDAproject", 
+                                             minDimsToUse = 15, npcs = 50, 
+                                             spikeGenes = NULL, nVariableFeatures = 2000, variableFeatureSelectionMethod = "vst",
+                                             tSNE_perplexity = 30, tSNE_maxIter = 2000,
+                                             UMAP_NumNeib = 40L, UMAP_MinDist = 0.2, UMAP_Seed = 1234, UMAP.NumEpoc = 500,
+                                             mean.cutoff = c(0.0125, 3), dispersion.cutoff = c(0.5, Inf)) 
+{
+  if (any(grepl("CD1", rownames(recomposedMat)))) {
+    print("Confirmed that genes are in rows.")
+  }
+  else {
+    recomposedMat <- t(recomposedMat)
+  }
   recomposedMat <- recomposedMat[, cellnames]
-  
   recomposedMat <- as(recomposedMat, "dgCMatrix")
-  # recomposedMat <- recomposedMat + 10 #abs(round(min(recomposedMat)))+1
-  metaDF <-  as.data.frame(metaDF[which(metaDF$Barcode %in% cellnames), ])
-  
+  metaDF <- as.data.frame(metaDF[which(metaDF$Barcode %in% 
+                                         cellnames), ])
   rownames(metaDF) <- metaDF$Barcode
-  #genes should be in rows and cells in columns. 
+  seuratObj <- CreateSeuratObject(counts = recomposedMat, 
+                                  min.features = 10, project = projectName, meta.data = metaDF)
   
-  # with imputed full matrix, create seurat object 
-  seuratObj <- CreateSeuratObject(counts = recomposedMat, min.features = 10, project = projectName, meta.data = metaDF)
+  print("Normalizing data:")
+  seuratObj <- NormalizeData(object = seuratObj, normalization.method = "LogNormalize", 
+                             #scale.factor = 10000,
+                             verbose = T)
   
-  #TODO: if the imputed is pre-normalized is this redundant?
-  seuratObj <- NormalizeData(seuratObj, normalization.method = "LogNormalize", scale.factor = 10000)
-  tempData <- as.data.frame(seuratObj@assays$RNA@data)
-  tempData[is.na(tempData)] <- 0
-  tempData <- as(as.matrix(tempData), "dgCMatrix")
-  seuratObj@assays$RNA@data <- tempData
+  #drop NaN values in matrix (otherwise DE test errors)    
```

Comment:
@kolabx why not do this on the recomposedMat, before creating the seurat object?  Do we know whether NAs are a problem for NormalizeData?

Reply:
sorry, i see this is @data, not @counts.  more to the point, do you know why NormalizeData is giving NAs on this input?  is that the real problem?

Reply:
I have a new update to deal with this, thus making this branch moot.